File: branches/LOFAR-Task7171/SAS/OTDB/src/setStatus.cc
//#  setStatus.cc: Force the status of an observation or pipeline
//#
//#  Copyright (C) 2004-2012
//#  ASTRON (Netherlands Foundation for Research in Astronomy)
//#  P.O.Box 2, 7990 AA Dwingeloo, The Netherlands, seg@astron.nl
//#
//#  This program is free software; you can redistribute it and/or modify
//#  it under the terms of the GNU General Public License as published by
//#  the Free Software Foundation; either version 2 of the License, or
//#  (at your option) any later version.
//#
//#  This program is distributed in the hope that it will be useful,
//#  but WITHOUT ANY WARRANTY; without even the implied warranty of
//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//#  GNU General Public License for more details.
//#
//#  You should have received a copy of the GNU General Public License
//#  along with this program; if not, write to the Free Software
//#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//#
//#  $Id: MACScheduler.cc 27964 2014-01-17 11:04:09Z mol $
#include <lofar_config.h>
#include <Common/LofarLogger.h>
#include <Common/ParameterSet.h>
#include "TreeStateConv.h"

using namespace LOFAR;
using namespace LOFAR::OTDB;

int main()
{
	std::string obsID;
	std::string status; // queued, active, completing, finished, aborted

	// Try to connect to the SAS database.
	ParameterSet* pParamSet = globalParameterSet();
	string username	= pParamSet->getString("OTDBusername");
	string DBname 	= pParamSet->getString("OTDBdatabasename");
	string password	= pParamSet->getString("OTDBpassword");
	string hostname	= pParamSet->getString("OTDBhostname");

	LOG_INFO_STR ("Trying to connect to the OTDB on " << hostname);
	conn = new OTDBconnection(username, password, DBname, hostname);

	LOG_INFO_STR ("Setting status of observation " << obsID << " to " << status);
	OTDB::TreeMaintenance tm(conn);
	TreeStateConv tsc(conn);
	tm.setTreeState(obsID, tsc.get(status));

	LOG_INFO_STR ("Disconnecting from OTDB");
	delete conn;
}

File: branches/LOFAR-Task8437/SAS/OTDB_Services/setStatus.py
#!/usr/bin/env python
#coding: iso-8859-15
#
# Copyright (C) 2015
# ASTRON (Netherlands Institute for Radio Astronomy)
# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
#
# This file is part of the LOFAR software suite.
# The LOFAR software suite is free software: you can redistribute it and/or
# modify it under the terms of the GNU General Public License as published
# by the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# The LOFAR software suite is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
#
# $Id$
"""

"""

from lofar.messaging.RPC import RPC
from lofar.sas.otdb.config import DEFAULT_OTDB_SERVICE_BUSNAME

def setStatus(obsid, status, otdb_busname=DEFAULT_OTDB_SERVICE_BUSNAME):
    with RPC("TaskSetStatus", busname=otdb_busname, timeout=10) as status_rpc:
        result, _ = status_rpc(OtdbID=obsid, NewStatus=status)

if __name__ == "__main__":
    from optparse import OptionParser
    import logging
    import sys

    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
    logger = logging.getLogger(__name__)

    # Check the invocation arguments
    parser = OptionParser("%prog -o obsid -s status [options]")
    parser.add_option("-B", "--busname", dest="busname", type="string", default=DEFAULT_OTDB_SERVICE_BUSNAME,
                      help="Busname on which OTDB commands are sent")
    parser.add_option("-o", "--obsid", dest="obsid", type="int", default=0,
                      help="Observation/tree ID to set status for")
    parser.add_option("-s", "--status", dest="status", type="string", default="",
                      help="New status")
    (options, args) = parser.parse_args()

    if not options.busname or not options.obsid or not options.status:
        parser.print_help()
        sys.exit(1)

    setStatus(options.obsid, options.status, otdb_busname=options.busname)
